sklearn/feature_extraction/hashing.py

```diff
@@ -112,12 +120,64 @@ def _validate_params(n_features, input_type):
         if input_type not in ("dict", "pair", "string"):
             raise ValueError("input_type must be 'dict', 'pair' or 'string',"
                              " got %r." % input_type)
+        if save_mappings not in (None, "both", "fit"):
+            raise ValueError("Unknown parameter passed to save_mappings: '{0}'"
+                             ". Valid parameters are 'fit', 'both' or None"
+                             .format(save_mappings))
 
-    def fit(self, X=None, y=None):
-        """No-op.
+    def _transform(self, raw_X, save_mappings):
+        raw_X = iter(raw_X)
+        if self.input_type == "dict":
+            raw_X = (_iteritems(d) for d in raw_X)
+        elif self.input_type == "string":
+            raw_X = (((f, 1) for f in x) for x in raw_X)
+
+        indices, indptr, values, feature_to_index_map_ = \
+            _hashing.transform(raw_X, self.n_features, self.dtype,
+                               self.alternate_sign, save_mappings)
+
+        if save_mappings:
+            self.feature_to_index_map_ = feature_to_index_map_
```

Comment:
Repeated calls to transform need to be updating this. Calls to fit or fit_transform need to reset it

Reply:
Yes, only when fit or always, but it still needs to be reset when fit_transform is called, and not reset when transform is repeatedly called.

Reply:
I think. This is sort of weird territory, because you insist we need to change things in transform.